### PR TITLE
Adds CreateCfp and CreateVoc DfTx

### PR DESCRIPTION
**Pull request by Jouzo**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

Adds CreateCfp and CreateVoc DfTx 

#### Which issue(s) does this PR fixes?:

Fixes part of  #517

#### Additional comments?:

CreateCfp and CreateVoc extend the same Composer CCreateProposal under the hood. 
This is done to match C++ implementation which has a single CCreatePropMessage to handle createcfp and createvoc (and will be used for type BlockRewardRellocation once implemented).

### CODE REVIEW

File: packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_governance_createcfp.test.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      status: governance.ProposalStatus.VOTING,
+      amount: createCfp.amount,
+      cyclesPaid: 1,
+      totalCycles: createCfp.cycles,
+      finalizeAfter: expect.any(Number),
+      payoutAddress: '2N5wvYsWcAWQUed5vfPxopxZtjkqoT8dFM3'
+    })
+  })
+
+  it('should reject with invalid type', async () => {
```
Comment (jingyi2811) [inline packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_governance_createcfp.test.ts:77]: Do you want to validate cycle?

'should not createCfp with cycles < 1'
'should not createCfp with cycle > 3'

Like what you did in your rpc?

File: packages/jellyfish-transaction-builder/src/txn/txn_builder_governance.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+    if (createVoc.cycles !== 2) {
+      throw new TxnBuilderError(TxnBuilderErrorType.INVALID_VOC_CYCLES,
+        'CreateVoc cycles should be 2'
+      )
+    }
+    return await this.createDeFiTx(
+      OP_CODES.OP_DEFI_TX_CREATE_VOC(createVoc),
+      changeScript,
+      new BigNumber('5') // For creation fee (regtest - 5, other than regtest - 50)
```
Comment (surangap) [inline packages/jellyfish-transaction-builder/src/txn/txn_builder_governance.ts:61]: take the network into consideration when passing the fee. i.e if regtest pass 5, other than that pass the relavant amount.

**Review by surangap — CHANGES_REQUESTED** [review]
request for some changes

File: packages/jellyfish-transaction-builder/src/txn/txn_builder_governance.ts (hunk 2 above)
Comment (Jouzo) [reply packages/jellyfish-transaction-builder/src/txn/txn_builder_governance.ts:61]: What is the right way to access and test on different network ?

Comment (Jouzo) [reply packages/jellyfish-transaction-builder/src/txn/txn_builder_governance.ts:61]: How would I know which network we are on here without major rewrite ?

File: packages/jellyfish-transaction/src/script/dftx/dftx_governance.ts [hunk 3]
```diff
@@ -102,3 +104,37 @@ export class CSetGovernance extends ComposableBuffer<SetGovernance> {
     ]
   }
 }
+
+export interface CreateProposal {
+  type: number // -------------| 1 byte, 0x01 | 0x02 | 0x03
```
Comment (jingyi2811) [inline packages/jellyfish-transaction/src/script/dftx/dftx_governance.ts:109]: 1 byte unsigned int, 0x01 | 0x02 | 0x03

**Review by surangap — DISMISSED** [review]
looks good.

File: packages/jellyfish-transaction/__tests__/script/dftx/dftx_governance/CreateCfp.test.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  it('should craft dftx with OP_CODES._()', () => {
+    const stack = [
+      OP_CODES.OP_RETURN,
+      OP_CODES.OP_DEFI_TX_CREATE_CFP(CreateProposal)
+    ]
+
+    const buffer = toBuffer(stack)
+    expect(buffer.toString('hex')).toBe(header + data)
```
Comment (jingyi2811) [inline packages/jellyfish-transaction/__tests__/script/dftx/dftx_governance/CreateCfp.test.ts:50]: Like what you advised me in my dftx PR. Change to toStrictEqual

File: packages/jellyfish-transaction-builder/src/txn/txn_builder_governance.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+export class TxnBuilderGovernance extends P2WPKHTxnBuilder {
+  /**
+   * Creates a Community fund proposal.
+   *
+   * @param {CreateProposal} createCfp txn to create
+   * @param {Script} changeScript to send unspent to after deducting the (converted + fees)
+   * @returns {Promise<TransactionSegWit>}
+   */
+  async createCfp (createCfp: CreateProposal, changeScript: Script, network = 'regtest'): Promise<TransactionSegWit> {
```
Comment (canonbrother) [inline packages/jellyfish-transaction-builder/src/txn/txn_builder_governance.ts:17]: can help change default reg to mainnet?? i will correct mine later
```suggestion
  async createCfp (createCfp: CreateProposal, changeScript: Script, network = 'mainnet'): Promise<TransactionSegWit> {
```

**Review by canonbrother — COMMENTED** [review]
overall good for me, just change the default network reg to mainnet will do

**Review by fuxingloh — DISMISSED** [review]

File: packages/jellyfish-transaction/src/script/dftx/dftx_governance.ts [hunk 6]
```diff
@@ -99,3 +101,37 @@ export class CSetGovernance extends ComposableBuffer<SetGovernance> {
     ]
   }
 }
+
+export interface CreateProposal {
```
Comment (ivan-zynesis) [inline packages/jellyfish-transaction/src/script/dftx/dftx_governance.ts:105]: Probably we can do this

```ts
interface Proposal { ... }
interface CreateCFP extends Proposal {
  type: 0x01
}
interface CreateVOC extends Proposal {
  type: 0x03
}
```

Comment (fuxingloh) [reply packages/jellyfish-transaction/src/script/dftx/dftx_governance.ts:105]: Actually yeah, let's do this instead

Comment (Jouzo) [reply packages/jellyfish-transaction/src/script/dftx/dftx_governance.ts:105]: Okay let me adjust it.
What about acronyms ? Should they always be uppercased ?

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**